apps/src/templates/sectionsRefresh/coteacherSettings/AddCoteacher.jsx

```diff
@@ -134,15 +113,36 @@ export default function AddCoteacher({
           }
         });
     },
-    [
-      setCoteachersToAdd,
-      setAddError,
-      inputValue,
-      setInputValue,
-      sectionId,
-      coteachersToAdd,
-      saveCoteacher,
-    ]
+    [addSavedCoteacher, setAddError, setInputValue]
+  );
+
+  const validateAndAddUnsavedCoteacher = useCallback(
+    email => {
+      getInputErrorMessage(email, coteachersToAdd, sectionId).then(
+        errorMessage => {
+          if (errorMessage === '') {
+            setCoteachersToAdd(existing => [email, ...existing]);
+            setInputValue('');
+          }
+          setAddError(errorMessage);
+        }
+      );
+    },
+    [setAddError, setInputValue, setCoteachersToAdd, coteachersToAdd, sectionId]
+  );
+
+  const handleAddEmail = useCallback(
+    e => {
+      e.preventDefault();
+      const newEmail = inputValue.trim();
+      if (!sectionId) {
+        validateAndAddUnsavedCoteacher(newEmail);
+      } else {
+        // Save coteacher only if we are editing an existing section.
+        return saveCoteacher(newEmail, sectionId);
```

Comment:
I think I was envisioning more that the API call decision could be made in the `getInputErrorMessage` method so that all the lovely error messaging you have for that code path could happen on editing an existing section too.